lib/backend.js

```diff
@@ -155,6 +166,13 @@ Backend.prototype.connect = function(connection, req) {
   // not used internal to ShareDB, but it is handy for server-side only user
   // code that may cache state on the agent and read it in middleware
   connection.agent = agent;
+
+  // Expose the DocPresence passed in through the constructor
+  // to the Doc class, which has access to the connection.
+  connection.DocPresence = this.presence.DocPresence;
```

Comment:
This will only work when the client and the server are on the same box. It's not going to work for the mainline case, where the client resides on a remote machine, and we can't directly access `Doc` from `Backend`.